code/modules/mob/living/simple_animal/hostile/terror_spiders/prince.dm

```diff
@@ -40,3 +42,35 @@
 	L.KnockDown(10 SECONDS)
 	L.adjustStaminaLoss(40)
 	return ..()
+
+/mob/living/simple_animal/hostile/poison/terror_spider/prince/Initialize(mapload)
+	. = ..()
+	if(mind)
+		var/obj/effect/proc_holder/spell/spell = new /obj/effect/proc_holder/spell/princely_charge()
+		mind.AddSpell(spell)
+	else
+		RegisterSignal(src, COMSIG_MOB_LOGIN, TYPE_PROC_REF(/mob/living/simple_animal/hostile/poison/terror_spider/prince, give_spell))
```

Comment:
I believe this is the signal you're actually looking for, since we're basing this on minds and not clients
```suggestion
		RegisterSignal(src, COMSIG_BODY_TRANSFER_TO, TYPE_PROC_REF(/mob/living/simple_animal/hostile/poison/terror_spider/prince, give_spell))
```

Reply:
As discussed in discord, the current one works fine. I also tested aghosting in and out of a prince, and it did not duplicate the spell!